### PR TITLE
Removed CoreBundle deprecations

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Removed
- Removed CoreBundle deprecations
```

### CODE REVIEW

File: src/Admin/CommentAdmin.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
 use Sonata\AdminBundle\Datagrid\ListMapper;
 use Sonata\AdminBundle\Form\FormMapper;
 use Sonata\AdminBundle\Form\Type\ModelListType;
-use Sonata\CoreBundle\Model\ManagerInterface;
+use Sonata\Doctrine\Model\ManagerInterface;
```
Comment (greg0ire) [inline src/Admin/CommentAdmin.php]: This is used in a signature, and signatures will not trigger autoload. Please trigger manually as described in the last paragraph of my blog post I recently added.

**Comment by core23** [comment]
> manual autoload is needed

?

**Comment by greg0ire** [comment]
See my comment above, you should trigger the autoload for the new type with a call to `class_exists` otherwise people extending the comment admin and using the legacy type in their signatures will get a crash.

**Comment by greg0ire** [comment]
More details here: https://twitter.com/greg0ire/status/1084933990811869190

File: src/Admin/CommentAdmin.php (hunk 1 above)
Comment (core23) [reply src/Admin/CommentAdmin.php]: Haven't done this on the other `Removed CoreBundle deprecations` PRs either...

Comment (core23) [reply src/Admin/CommentAdmin.php]: I also didn't deprecate anything in this PR. I removed the deprecations caused be the CoreBundle

Comment (core23) [reply src/Admin/CommentAdmin.php]: ping @greg0ire

Comment (greg0ire) [reply src/Admin/CommentAdmin.php]: Yeah those are not used in type hints in most cases, unlike here.

Comment (core23) [reply src/Admin/CommentAdmin.php]: So, what to do?

Comment (core23) [reply src/Admin/CommentAdmin.php]: This looks wrong. This trigger should be added to the `doctrine-extensions` class. 

In fact there is a trigger for this: https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Model/ManagerInterface.php#L101

Comment (greg0ire) [reply src/Admin/CommentAdmin.php]: There is indeed, but that file will not be loaded because of a type hint. Type hints do not trigger autoload.

Comment (greg0ire) [reply src/Admin/CommentAdmin.php]: Wait no it seems to work :thinking:

Comment (greg0ire) [reply src/Admin/CommentAdmin.php]: I don't get why, the parent interface gets autoloaded by something… guess things are fine then

**Comment by greg0ire** [comment]
Thanks @core23 , and apologies for the blocking, I should have tried first! (but still, I don't understand why the interfaces are autoloaded)

**Pull request merged**